server/api/crud/model_monitoring/deployment.py

```diff
@@ -1005,48 +1006,77 @@ def check_if_credentials_are_set(self, only_project_secrets: bool = False):
             ]
         ):
             return
+        if with_upgrade_case_check:
+            with_upgrade_case_check = self._set_credentials_after_server_upgrade()
+
+        if not with_upgrade_case_check:
+            raise mlrun.errors.MLRunBadRequestError(
+                "Model monitoring credentials are not set. "
+                "Please set them using the set_model_monitoring_credentials API/SDK "
+                "or pass fetch_credentials_from_sys_config=True when using enable_model_monitoring API/SDK."
+            )
+
+    def _set_credentials_after_server_upgrade(self) -> bool:
+        """
+        Check and Set the model monitoring credentials for old project that had model monitoring on it before the server
+        upgrade.
+        :return: True if the credentials are set, otherwise False.
+        """
+        # Check if:
+        #   1. There is model endpoints.
+        #   2. Stream pod is on
+        #   3. There is already part of the secrets set.
+        #  if True, set the cred in to the project secret (from exist cred/ from sys config / v3io by default).
```

Comment:
Just a recommendation to make it clearer

```suggestion
    def _set_credentials_after_server_upgrade(self) -> bool:
        """
        Check and set the model monitoring credentials for old project that included model monitoring before the server
        upgrade. Will set the credentials only if at least one of the following conditions is met:
        1. There is at least one model endpoint
        2. Model monitoring stream pod is running
        3. Part of the model monitoring credentials are already set

        If True, set the cred in to the project secret (from exist cred/from sys config/v3io by default).

        :return: True if the credentials are set, otherwise False.
        """

```